logic/path/src/selectors/dfs.rs

```diff
@@ -38,13 +58,12 @@
         if other.fully_explored == self.fully_explored {
             match self.path.len().cmp(&other.path.len()) {
                 Ordering::Equal => self.weight.cmp(&other.weight),
-                Ordering::Greater => Ordering::Greater,
-                Ordering::Less => Ordering::Less,
+                o => o,
             }
         } else if other.fully_explored {
-            Ordering::Less
-        } else {
             Ordering::Greater
+        } else {
+            Ordering::Less
```

Comment:
No, it is actually correct. 
If `other.fully_explored`, the instance at hand is considered greater, because it is unexplored and therefore should be on top of the max-heap.